### ISSUE TITLE
Crash on installing 0.11.0 with setup.py

### ISSUE THREAD

**Issue by jkotan** [pyqtgraph/pyqtgraph#1271]
# Short description
I want to test the newest version of pyqtgraph tagged as pyqtgraph-0.11.0  but it is not possible to install it because of a bug in setupHelper.py

### Code to reproduce

```bash
git checkout -b pyqtgraph-0.11.0 bpyqtgraph-0.11.0
python setup.py install --prefix=$HOME/testinst/
```

### Expected behavior

I would expect that pyqtgraph will be installed at `$HOME/testinst/`
### Real behavior
Calling setup.py crashes because of lack of the snapshot local version.
```
~/sources/pyqtgraph$ python setup.py install --prefix=$HOME/testinst/
Traceback (most recent call last):
  File "setup.py", line 70, in <module>
    version, forcedVersion, gitVersion, initVersion = helpers.getVersionStrings(pkg='pyqtgraph')
  File "tools/setupHelpers.py", line 470, in getVersionStrings
    _, local = gitVersion.split('+')
ValueError: need more than 1 value to unpack
```

### Tested environment(s)

 * PyQtGraph version: 0.11.0
 * Qt Python binding: 5.11.3
 * Python version:  2.7.16
 * NumPy version: 1.16.2
 * Operating system:  debian buster
 * Installation method: git

**Comment by 2xB** [comment]
Additional information: This happens when checking out `pyqtgraph-0.11.0` and can be tested with the following code, executed in the top-level directory of the git repository:
```python3
import os, sys, re
path = os.path.split(__file__)[0]
sys.path.insert(0, os.path.join(path, 'tools'))
import setupHelpers as helpers

a = helpers.getVersionStrings(pkg='pyqtgraph')
```

**Comment by ksunden** [comment]
That test works fine on my machine (on python3, so maybe something py2 specific?) I mean I did the release from that tag, so I knew it worked then...

**Comment by j9ac9k** [comment]
I have pyenv installed so I'll try this out on a variety of different python versions and report back shortly.

**Comment by ksunden** [comment]
I also have pyenv, and tried it out on 2.7.16 since posting, still works fine for me...

**Comment by ksunden** [comment]
As a result I looked and saw that our wheel is py3 only, so uploading the 0.11.0 py2 wheel (the sdist is there anyway, so can be installed, but wheels are nice)

**Comment by 2xB** [comment]
@ksunden I use Python3.8... Interesting. So for me, the issue is that `gitVersion` as returned by `getGitVersion` is only `0.11.0`, when I'm on `pyqtgraph-0.11.0`, while the line that is erroring actually expects something like `0.11.0+g12a7c44`, which I get when being on an actual branch. So what's the output of the following code for you, when you are on `pyqtgraph-0.11.0`?

```python3
import os, sys, re
path = os.path.split(__file__)[0]
sys.path.insert(0, os.path.join(path, 'tools'))
import setupHelpers as helpers

print(helpers.getGitVersion(tagPrefix='pyqtgraph'+'-'))
```

**Comment by j9ac9k** [comment]
Definitely not a py2/py3 thing, doing the steps in the OP (with one correction) I'm able to replicate the issue

```bash
pyqtgraph-27 ❯ git checkout -b pyqtgraph-0.11.0 pyqtgraph-0.11.0
Switched to a new branch 'pyqtgraph-0.11.0'

~/Developer/pyqtgraph pyqtgraph-0.11.0*
pyqtgraph-27 ❯ python setup.py install --prefix=$HOME/testinst/
Traceback (most recent call last):
  File "setup.py", line 70, in <module>
    version, forcedVersion, gitVersion, initVersion = helpers.getVersionStrings(pkg='pyqtgraph')
  File "tools/setupHelpers.py", line 470, in getVersionStrings
    _, local = gitVersion.split('+')
ValueError: need more than 1 value to unpack

~/Developer/pyqtgraph pyqtgraph-0.11.0*
pyqtgraph-27 ❯
```

**Comment by ksunden** [comment]
I think I know why it is working for me... I have all of my git repos as git submodules for my dev setup, so it isn't finding the current branch _ever_ (which is fine when I'm releasing, as that gets stripped intentionally then).

For me `.git` is a file which points to my master repo's `.git` folder...

**Comment by j9ac9k** [comment]
> I think I know why it is working for me... I have all of my git repos as git submodules for my dev setup, so it isn't finding the current branch _ever_ (which is fine when I'm releasing, as that gets stripped intentionally then).
> 
> For me `.git` is a file which points to my master repo's `.git` folder...

First thing I thought of while reading that:  https://xkcd.com/1172/

Joking aside, would the fix here to either do a regex match or something like `str.rpartition(str.lpartition())`?

**Comment by ksunden** [comment]
Except the opposite here, my workflow was blissfully unaware of a problem

**Comment by ixjlyons** [comment]
Fwiw I think installing from git via pip is also broken for the same reason #1150

**Comment by j9ac9k** [comment]
This should have been fixed with #1402
